sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/handler/LdsHandler.java

```diff
@@ -0,0 +1,112 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Based on dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/protocol/impl/LdsProtocol.java
+ * from the Apache dubbo project.
+ */
+
+package io.sermant.implement.service.xds.handler;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import io.envoyproxy.envoy.config.listener.v3.Listener;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+import io.grpc.stub.StreamObserver;
+import io.sermant.implement.service.xds.cache.XdsDataCache;
+import io.sermant.implement.service.xds.client.XdsClient;
+import io.sermant.implement.service.xds.env.XdsConstant;
+import io.sermant.implement.service.xds.utils.LdsProtocolTransformer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.logging.Level;
+
+/**
+ * LdsHandler
+ *
+ * @author daizhenyu
+ * @since 2024-07-30
+ **/
+public class LdsHandler extends XdsHandler {
+    /**
+     * construction method
+     *
+     * @param client xds client
+     */
+    public LdsHandler(XdsClient client) {
+        super(client);
+        this.resourceType = XdsConstant.LDS_RESOURCE_TYPE;
+    }
+
+    @Override
+    protected void handleResponse(String requestKey, DiscoveryResponse response) {
+        Set<String> oldRouteResources = XdsDataCache.getRouteResources();
+        XdsDataCache.updateHttpConnectionManagers(LdsProtocolTransformer
+                .getHttpConnectionManager(decodeResource2Listener(response)));
+
+        // send ack
+        StreamObserver<DiscoveryRequest> requestObserver = XdsDataCache.getRequestObserver(requestKey);
+        requestObserver.onNext(builtAckDiscoveryRequest(response, Collections.EMPTY_SET));
+
+        updateRdsSubscription(oldRouteResources);
+    }
+
+    private List<Listener> decodeResource2Listener(DiscoveryResponse response) {
+        List<Listener> listeners = new ArrayList<>();
+        for (Any any : response.getResourcesList()) {
+            try {
+                if (any == null) {
+                    continue;
+                }
+                listeners.add(any.unpack(Listener.class));
+            } catch (InvalidProtocolBufferException e) {
+                LOGGER.log(Level.SEVERE, "Decode resource to Listener failed.", e);
+            }
+        }
+        return listeners;
+    }
```

Comment:
It is recommended that a common method be proposed to unpack.

Reply:
Modified to use abstract methods